src/crimson/os/seastore/transaction.h

```diff
@@ -83,7 +84,13 @@ class Transaction {
     if (is_weak()) return;
 
     auto [iter, inserted] = read_set.emplace(this, ref);
-    ceph_assert(inserted);
+    if (!inserted) {
+      ::crimson::get_logger(ceph_subsys_seastore).debug(
+	"{} {} {} already in read_set",
+	__func__,
+	(void*)this,
+	*ref);
+    }
```

Comment:
I changed the `SegmentCleander::scan_extents` to used `parallel_for_each`

Reply:
I don't see a SegmentCleaner::scan_extents.  Journal::scan_extents doesn't use parallel_for_each.  I think you mean SegmentCleaner::gc_reclaim_space or TransactionManager::rewrite_extents.  Both of those were introduced in
"crimson/os/seastore: use extent placement manager to alloc ondisk space for written blocks".  I'm not convinced that there's an advantage to performing those operations in parallel just yet, and if there is it would still require throttling the number of concurrent operations.  There's also really no reason to address that change as part of this already complicated PR.  Please back out those portions.

Reply:
Will do